Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.cpp

```diff
@@ -1119,6 +1074,240 @@ void GenericConstraintProblem::unbuiltGaussSeidel(SReal timeout, GenericConstrai
     }
 }
 
+void GenericConstraintProblem::NNCG(GenericConstraintSolver* solver, int iterationNewton)
+{
+    if(!solver)
+        return;
+
+    const int dimension = getDimension();
+
+    if(!dimension)
+    {
+        currentError = 0.0;
+        currentIterations = 0;
+        return;
+    }
+
+    m_lam.clear();
+    m_lam.resize(dimension);
+    m_deltaF.clear();
+    m_deltaF.resize(dimension);
+    m_deltaF_new.clear();
+    m_deltaF_new.resize(dimension);
+    m_p.clear();
+    m_p.resize(dimension);
+
+
+    SReal *dfree = getDfree();
+    SReal *force = getF();
+    SReal **w = getW();
+    SReal tol = tolerance;
+
+    SReal *d = _d.ptr();
+
+
+    SReal error = 0.0;
+    bool convergence = false;
+    sofa::type::vector<SReal> tempForces;
+
+    if(sor != 1.0)
+    {
+        tempForces.resize(dimension);
+    }
+
+    if(scaleTolerance && !allVerified)
+    {
+        tol *= dimension;
+    }
+
+    for(int i=0; i<dimension; )
+    {
+        if(!constraintsResolutions[i])
+        {
+            msg_error("GenericConstraintSolver") << "Bad size of constraintsResolutions in GenericConstraintProblem" ;
+            break;
+        }
+        constraintsResolutions[i]->init(i, w, force);
+        i += constraintsResolutions[i]->getNbLines();
+    }
+
+    sofa::type::vector<SReal> tabErrors(dimension);
+    bool constraintsAreVerified = true;
```

Comment:
Move this variable in the next scope. You can make it `static constexpr` also. Also valid for the other ` constraintsAreVerified` variables